src/test/java/uk/gov/companieshouse/service/AppealServiceTest.java

```diff
@@ -133,7 +133,39 @@ public void testGetAppealById_returnsEmptyAppeal() {
 
         when(appealRepository.findById(any(String.class))).thenReturn(Optional.empty());
 
-        final Optional<Appeal> appealOpt = appealService.getAppeal(TEST_RESOURCE_ID);
+        final Optional<Appeal> appealOpt = appealService.getAppealById(TEST_RESOURCE_ID);
+
+        assertFalse(appealOpt.isPresent());
+        assertEquals(Optional.empty(), appealOpt);
+    }
+
+    @Test
+    public void testGetAppealByPenaltyReference_returnsAppeal() {
+
+        final Appeal validAppeal = getValidAppeal();
+
+        when(appealRepository.findByPenaltyReference(any(String.class),any(String.class))).thenReturn(Optional.of(validAppeal));
```

Comment:
I'm not convinced here you've actually tested that the service query actually works because you've mocked the whole call. I would suggest writing a test to verify that the query does what it is intended

Reply:
We only mocked the call to the appeal repository which is what makes the query to the database.

Reply:
yes but that's also a mock. What I'm saying is that you haven't tested `"{ 'penaltyIdentifier' : {'companyNumber' : ?0, 'penaltyReference' : ?1 } }"`. How do we know that this actually works?